File: exercice2_5.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/types.h>
#include <sys/wait.h>
#include <time.h>
#include <errno.h>

int main()
{
	srand(time(NULL)) ;
	time_t debut, fin ;
	debut = time(NULL) ;

	pid_t fils;
	int i = 0 ;
	for (i = 1; i <= 2; ++i)
	{
		fils = fork() ;
		if (fils == -1)
		{
			perror("Erreur de fork\n") ;
			exit(errno) ;
		}
		else if (fils == 0)
		{
			int temps = (rand() % 10) + 1 ;
			printf("je suis le fils %d\n", i);
			sleep(temps) ;
			printf("fils %d terminé après %d second\n", i, temps);
			return (0) ;
		}
	}
	for (i = 1; i <= 2; ++i)
	{
		wait(NULL) ;
	}
	fin = time(NULL) ;
	printf("La durée totale d'execution est %ld\n", fin - debut);
	return (0) ;
}

File: Python/exercice2_5.py
import os
import sys
import time
import random
import signal

def main():
   
    random.seed(int(time.time()))
    debut = time.time()
    
    for i in range(1, 3):
        pid_fils = os.fork()
        
        if pid_fils == -1: 
            print(f"Erreur de fork pour fils {i}")
            sys.exit(1)
        
        elif pid_fils == 0: 
            temps = random.randint(1, 10)
            print(f"je suis le fils {i}")
            
            time.sleep(temps)
            
            print(f"fils {i} terminé après {temps} secondes")
            return 
    
    for i in range(1, 3):
        os.wait() ;
    
    fin = time.time()
    print(f"La durée totale d'exécution est {int(fin - debut)} secondes")

if __name__ == "__main__":
    main()